### ISSUE TITLE
 Resizing the window in the  game of life example divides the cell automata into several independent areas.

### ISSUE THREAD

**Issue by palsol** [cprogrammer1994/ModernGL#121]
### Expected behavior

If you change the size of the window to values other than the degree of two, the cellular automaton works as a unit in the whole area.

![image](https://user-images.githubusercontent.com/10175363/27761229-7bd90f44-5e60-11e7-8c59-22cbdf99f7b4.png)

### Actual behavior

The cellular automaton is divided into several independent regions.

![image](https://user-images.githubusercontent.com/10175363/27761217-694590be-5e60-11e7-912c-04438a5f7261.png)

### Steps to reproduce behavior
```python
import struct
import numpy

import GLWindow
import ModernGL

windowSize = 511
pixels = numpy.round(numpy.random.rand(windowSize, windowSize)).astype('float32')
grid = numpy.dstack(numpy.mgrid[0:windowSize, 0:windowSize][::-1] / windowSize).astype('float32')

wnd = GLWindow.create_window(windowSize, windowSize)
ctx = ModernGL.create_context()

prog = ctx.program([
    ctx.vertex_shader('''
        #version 330
        in vec2 vert;
        out vec2 text;
        void main() {
            text = vert;
            gl_Position = vec4((vert * 2.0 - 1.0) * 1, 0.0, 1.0);
        }
    '''),
    ctx.fragment_shader('''
        #version 330
        uniform sampler2D Texture;
        in vec2 text;
        out vec4 color;
        void main() {
            color = texture(Texture, text);
        }
    '''),
])

trans = ctx.program(
    ctx.vertex_shader('''
        #version 330
        uniform sampler2D Texture;
        uniform int windowSize;
        in vec2 text;
        out float vert;
        #define LIVING 0.0
        #define DEAD 1.0
        void main() {
            bool living = texture(Texture, text).r < 0.5;
            int neighbours = 0;
            vec2 d = vec2(1.0/windowSize, 1.0/windowSize);
            if (texture(Texture, text + vec2(-d.x, -d.y)).r < 0.5) neighbours++;
            if (texture(Texture, text + vec2(-d.x, 0.0)).r < 0.5) neighbours++;
            if (texture(Texture, text + vec2(-d.x, d.y)).r < 0.5) neighbours++;
            if (texture(Texture, text + vec2(d.x, -d.y)).r < 0.5) neighbours++;
            if (texture(Texture, text + vec2(d.x, 0.0)).r < 0.5) neighbours++;
            if (texture(Texture, text + vec2(d.x, d.y)).r < 0.5) neighbours++;
            if (texture(Texture, text + vec2(0.0, d.y)).r < 0.5) neighbours++;
            if (texture(Texture, text + vec2(0.0, -d.y)).r < 0.5) neighbours++;
            if (living) {
                vert = (neighbours == 2 || neighbours == 3) ? LIVING : DEAD;
            } else {
                vert = (neighbours == 3) ? LIVING : DEAD;
            }
        }
    '''),
    varyings=["vert"]
)

trans.uniforms['windowSize'].value = windowSize

texture = ctx.texture((windowSize, windowSize), 1, pixels.tobytes(), floats=True)
texture.filter = ModernGL.NEAREST
texture.swizzle = 'RRR1'
texture.use()

vbo = ctx.buffer(struct.pack('8f', 0, 0, 0, 1, 1, 0, 1, 1))
vao = ctx.simple_vertex_array(prog, vbo, ['vert'])

text = ctx.buffer(grid.tobytes())
tao = ctx.simple_vertex_array(trans, text, ['text'])
pbo = ctx.buffer(reserve=pixels.nbytes)

while wnd.update():
    ctx.viewport = wnd.viewport

    tao.transform(pbo)
    texture.write(pbo)

    vao.render(ModernGL.TRIANGLE_STRIP)
```

### Version info

```
ModernGL: 4.1.7
Vendor: NVIDIA Corporation
Renderer: GeForce GTX 960/PCIe/SSE2
Version: 4.5.0 NVIDIA 382.05
Python: 3.5.3 |Continuum Analytics, Inc.| (default, May 15 2017, 10:43:23) [MSC v.1900 64 bit (AMD64)]
Platform: win32
```

**Comment by cprogrammer1994** [comment]
Only works with `width = 2ⁿ` and `height = 2ⁿ`

**Comment by cprogrammer1994** [comment]
Solved in #124
